index.d.ts

```diff
@@ -23,7 +23,7 @@ export interface FastifyBasicAuthOptions {
     req: FastifyRequest,
     reply: FastifyReply,
     done: (err?: Error) => void
-  ): void | Promise<void>;
+  ): Error | void | Promise<Error | void>;
```

Comment:
I do not think it should return `Error` when it is using `callback`.
```suggestion
  ): void | Promise<Error | void>;
```